h/static/scripts/time.js

```diff
@@ -6,43 +6,79 @@ var day = hour * 24;
 var month = day * 30;
 var year = day * 365;
 
+function lessThanThirtySecondsAgo(date, now) {
+  return ((now - date) < 30000);
+}
+
+function lessThanOneMinuteAgo(date, now) {
+  return ((now - date) < 60000);
+}
+
+function lessThanOneHourAgo(date, now) {
+  return ((now - date) < (60 * 60 * 1000));
+}
+
+function lessThanOneDayAgo(date, now) {
+  return ((now - date) < (24 * 60 * 60 * 1000));
+}
+
+function thisYear(date, now) {
+  return date.getFullYear() === now.getFullYear();
+}
+
+function delta(date, now) {
+  return Math.round((now - date) / 1000);
+}
+
+function nSec(date, now) {
+  return '{} sec'.replace('{}', Math.floor(delta(date, now)));
+}
+
+function nMin(date, now) {
+  return '{} min'.replace('{}', Math.floor(delta(date, now) / minute));
+}
+
+function nHr(date, now) {
+  return '{} hr'.replace('{}', Math.floor(delta(date, now) / hour));
+}
+
+function dayAndMonth(date) {
+  return date.toLocaleDateString(undefined, {day: '2-digit', month: 'short'});
+}
+
+function dayAndMonthAndYear(date) {
+  return date.toLocaleDateString(
+    undefined, {day: '2-digit', month: 'short', year: 'numeric'});
+}
+
 var BREAKPOINTS = [
-  [30,         'moments ago',    1],
-  [minute,     '{} seconds ago', 1],
-  [2 * minute, 'a minute ago',   minute],
-  [hour,       '{} minutes ago', minute],
-  [2 * hour,   'an hour ago',    hour],
-  [day,        '{} hours ago',   hour],
-  [2 * day,    'a day ago',      day],
-  [month,      '{} days ago',    day],
-  [year,       '{} months ago',  month],
-  [2 * year,   'one year ago',   year],
-  [Infinity,   '{} years ago',   year]
+  [lessThanThirtySecondsAgo,    function () {return 'Just now';},      1],
```

Comment:
The last item in each tuple is now redundant. I think this would be clearer to read if written as an array of formatters:

```js
var formatters = [
  {test: testFunc, format: formatFunc},
  {test: testFunc2, format: formatFunc2},
  ...
];
```